### PR TITLE
[[ DontUseQT ]] Change and document the default settings for dontUseQT

### PR DESCRIPTION
**Pull request by montegoulding**

### CODE REVIEW

File: docs/dictionary/property/dontUseQT.lcdoc [hunk 1]
```diff
@@ -16,24 +18,38 @@ Example:
...
-        * Displaying a player object
-        * Playing an audio clip or video clip
-        * Recording sound using the record sound command
-        * Using the recordFormats <function> or the <answer record> command
-        * Using a visual effect with visual effect, <show>, <hide>, or unlock screen
-        * Using the answer effect <command> or <QTEffects> function
-        * Using the <QTVersion> function
+On Windows, when <dontUseQT> is true players will use the DirectShow API
+for multimedia playback.
+
```
Comment (livecodefraser) [inline docs/dictionary/property/dontUseQT.lcdoc]: The QT player is gone entirely from Windows in 8.1 and the dontUseQT property is ignored for players (both the global and the player-local properties).

File: docs/notes/feature-dontuseqt_for_windows_player.md [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
-greater or equal to 10.8. This means that any player object created will use the AVFoundation
-API for multimedia playback. With this new feature, you can set the dontUseQT property of a
-player to false, without changing the value of the global dontUseQt property. In that way you 
-can have both QuickTime and AVFoundation players playing at the same time. This can be particular 
-useful for supporting some media formats or codecs that are not supported by the default AVFoundation player
-(for example .midi files, Sorenson Video 3, H.261 codecs etc) 
+On Windows, the default value of the `dontUseQt` and `dontUseQtEffects`
+(global) properties have changed from true to false. This means that by
+default players created on Windows will use the DirectShow API for
+multimedia playback.
```
Comment (livecodefraser) [inline docs/notes/feature-dontuseqt_for_windows_player.md:9]: As above, all players on Windows will use DirectShow.

**Comment by livecodefraser** [comment]
This PR is making me wonder whether we should just ignore attempts to change dontUseQT on Windows... writes would be ignored (without an error) and reads would always return true.

**Comment by peter-b** [comment]
@livecodefraser I like that idea.

**Comment by montegoulding** [comment]
@livecodefraser @peter-b I think I was confused because `FEATURE_QUICKTIME` is still defined for Windows. Are you suggesting changing the value has no effect or that it should remain true no matter what the user does? I think it may still be feasible to use QT effects should we wish to allow it.

BTW I believe the following should be done now along with the docs changes you mentioned:
- Not define `FEATURE_QUICKTIME` on Windows.
- I believe `FEATURE_QUICKTIME_EFFECTS` can be separate. It may (probably is) better to stop supporting QT entirely on Windows now in which case we should also not define `FEATURE_QUICKTIME_EFFECTS` for Windows. If we do that we should:
  - remove `quicktime.cpp` and the quicktime stubs phase from the Windows build
  - remove mention of QuickTime sdk from `INSTALL-win.md`
- Change the following so it checks for `FEATURE_QUICKTIME_EFFECTS` rather than `FEATURE_PLATFORM_PLAYER` && `FEATURE_QUICKTIME` and then init via `if (!MCQTInitialize())`. At the moment I suspect this means QT effects isn't working (even on an older Mac) if you haven't done something else to ensure QT is loaded.

```
#ifndef FEATURE_PLATFORM_PLAYER
#if defined(FEATURE_QUICKTIME)
    // MW-2010-07-07: Make sure QT is only loaded if we actually are doing an effect
    if (t_effects != nil)
        if (!MCdontuseQTeffects)
            if (!MCtemplateplayer -> isQTinitted())
                MCtemplateplayer -> initqt();
#endif  
#endif
```

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay 3e8ec62

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok 3e8ec6236fcb8ab4a170f2083d4558ba55b67d48

**Comment by livecode-vulcan** [comment]
😞 test failure 3e8ec6236fcb8ab4a170f2083d4558ba55b67d48
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2629)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2639)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2624)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2640)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2646)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2652)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2664)

**Comment by livecodefraser** [comment]
From the build log: 

kernel.lib(w32dcw32.obj) : error LNK2019: unresolved external symbol "void __
cdecl MCQTHandleRecord(void)" (?MCQTHandleRecord@@YAXXZ) referenced in function
 "public: unsigned char __thiscall MCScreenDC::handle(double,unsigned char,unsigned char,unsigned char &,unsigned char &)" (?handle@MCScreenDC@@QAEENEEAAE0@Z)

**Comment by montegoulding** [comment]
Hmm... must be something not `#ifdefed` correctly. I'll investigate.

**Comment by montegoulding** [comment]
That should fix it @livecodefraser

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay 403c06f

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok 403c06f1d10a9255ea2bbc6af385af036d2827ef

**Comment by livecode-vulcan** [comment]
😎 test success 403c06f1d10a9255ea2bbc6af385af036d2827ef
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2633)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2643)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2628)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2644)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2650)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2656)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2668)

**Pull request merged**